backend/Modules/Users/HitMeApp.Users.Tests.Unit/Core/UserTests.cs

```diff
@@ -0,0 +1,242 @@
+
+            _user.DomainEvents.Count().ShouldBe(1);
+            var domainEvent = _user.DomainEvents.First();
+            var userAddedPreference = domainEvent.ShouldBeOfType<UserAddedPreference>();
+            userAddedPreference.UserId.ShouldBe(_user.Id);
+            userAddedPreference.Preference.ShouldBe(newTrait);
+            _user.Preferences.First().ShouldBe(newTrait);
+        }
+
+        [Fact]
+        public void add_preference_should_not_take_effect_if_preference_was_previously_added()
+        {
+            var preferenceToAdd = Trait.New("Test");
+            _user.AddPreference(preferenceToAdd);
+            _user.ClearEvents();
+
+            var loadedPreferenceToAdd = Trait.Load(preferenceToAdd.Id, preferenceToAdd.Value);
+            _user.AddPreference(loadedPreferenceToAdd);
+
+            _user.DomainEvents.Count().ShouldBe(0);
+        }
+
+        [Fact]
+        public void remove_preference_should_take_effect_if_preferece_was_previously_added()
+        {
+            var preferenceToRemove = Trait.New("Test");
+            _user.AddPreference(preferenceToRemove);
+            _user.ClearEvents();
+
+            var loadedPreferenceToRemove = Trait.Load(preferenceToRemove.Id, preferenceToRemove.Value);
+            _user.RemovePreference(loadedPreferenceToRemove);
+
+            _user.DomainEvents.Count().ShouldBe(1);
+            var domainEvent = _user.DomainEvents.First();
+            var userRemovedPreference = domainEvent.ShouldBeOfType<UserRemovedPreference>();
+            userRemovedPreference.UserId.ShouldBe(_user.Id);
+            userRemovedPreference.Preference.Id.ShouldBe(loadedPreferenceToRemove.Id);
+            userRemovedPreference.Preference.Value.ShouldBe(loadedPreferenceToRemove.Value);
+        }
+
+        [Fact]
+        public void remove_preference_should_not_take_effect_if_preference_was_not_previously_added()
+        {
+            _user.RemovePreference(Trait.New("Test"));
+
+            _user.DomainEvents.Count().ShouldBe(0);
+        }
+
+        [Fact]
+        public void remove_preference_should_only_remove_requested_preference()
+        {
+            var preferenceToRemove = Trait.New("Test2");
+            _user.AddPreference(Trait.New("Test"));
+            _user.AddPreference(preferenceToRemove);
+            _user.AddPreference(Trait.New("Test3"));
+            _user.ClearEvents();
+
+            var loadedPreferenceToRemove = Trait.Load(preferenceToRemove.Id, preferenceToRemove.Value);
+            _user.RemovePreference(loadedPreferenceToRemove);
+
+            _user.Preferences.Count().ShouldBe(2);
+            _user.DomainEvents.Count().ShouldBe(1);
+            var domainEvent = _user.DomainEvents.First();
+            var userRemovedPreference = domainEvent.ShouldBeOfType<UserRemovedPreference>();
+            userRemovedPreference.UserId.ShouldBe(_user.Id);
+            userRemovedPreference.Preference.ShouldBe(preferenceToRemove);
+        }
```

Comment:
Should we maybe verify if these 2 preferences that are left are in fact "Test" and "Test3"?